compose/service.py

```diff
@@ -306,14 +302,19 @@ def get_net_name(self):
         else:
             return
 
-    def _next_container_name(self, all_containers, one_off=False):
-        bits = [self.project, self.name]
-        if one_off:
-            bits.append('run')
-        return '_'.join(bits + [str(self._next_container_number(all_containers))])
-
-    def _next_container_number(self, all_containers):
-        numbers = [parse_name(c.name).number for c in all_containers]
+    def get_container_name(self, number, one_off=False):
+        # TODO: Implement issue #652 here
+        return build_container_name(self.project, self.name, number, one_off)
+
+    # TODO: this would benefit from github.com/docker/docker/pull/11943
+    # to remove the need to inspect every container
```

Comment:
Not very familiar with python, but does this currently require to inspect *every* container, or the containers after filtering by project (and service name)? Trying to determine how important https://github.com/docker/docker/pull/11943 is for this.